lib/cisco_node_utils/interface.rb

```diff
@@ -1019,9 +1019,19 @@ def default_switchport_mode_private_vlan_host_association
     end
 
     def prepare_array(is_list)
```

Comment:
It would be helpful to include some comments for this increasingly complicated method; i.e. some sample inputs and the expected resulting format. Also note for future methods (I'm not going to make you change it here), but please try to choose better method and variable names. `prepare_array` is fairly vague when we use arrays all over the place; `is_list` isn't really the 'is' list when you use this method to format the `should_list`; etc. Try to code for the people that will have to debug this or extend the functionality down the road.